processing/src/main/java/org/apache/druid/query/filter/RangeFilter.java

```diff
@@ -0,0 +1,1025 @@
+              return false;
+            }
+            final int lowerComparing = stringComparator.compare(input, lowerBound);
+            final int upperComparing = stringComparator.compare(upperBound, input);
+            return ((lowerComparing > 0)) && (upperComparing > 0);
+          };
+        } else if (lowerStrict) {
+          return input -> {
+            if (NullHandling.isNullOrEquivalent(input)) {
+              return false;
+            }
+            final int lowerComparing = stringComparator.compare(input, lowerBound);
+            final int upperComparing = stringComparator.compare(upperBound, input);
+            return (lowerComparing > 0) && (upperComparing >= 0);
+          };
+        } else if (upperStrict) {
+          return input -> {
+            if (NullHandling.isNullOrEquivalent(input)) {
+              return false;
+            }
+            final int lowerComparing = stringComparator.compare(input, lowerBound);
+            final int upperComparing = stringComparator.compare(upperBound, input);
+            return (lowerComparing >= 0) && (upperComparing > 0);
+          };
+        } else {
+          return input -> {
+            if (NullHandling.isNullOrEquivalent(input)) {
+              return false;
+            }
+            final int lowerComparing = stringComparator.compare(input, lowerBound);
+            final int upperComparing = stringComparator.compare(upperBound, input);
+            return (lowerComparing >= 0) && (upperComparing >= 0);
+          };
+        }
+      } else if (hasUpperBound()) {
+        if (upperStrict) {
+          return input -> {
+            if (NullHandling.isNullOrEquivalent(input)) {
+              return false;
+            }
+            final int upperComparing = stringComparator.compare(upperBound, input);
+            return upperComparing > 0;
+          };
+        } else {
+          return input -> {
+            if (NullHandling.isNullOrEquivalent(input)) {
+              return false;
+            }
+            final int upperComparing = stringComparator.compare(upperBound, input);
+            return upperComparing >= 0;
+          };
+        }
+      } else if (hasLowerBound()) {
+        if (lowerStrict) {
+          return input -> {
+            if (NullHandling.isNullOrEquivalent(input)) {
+              return false;
+            }
+            final int lowerComparing = stringComparator.compare(input, lowerBound);
+            return lowerComparing > 0;
+          };
+        } else {
+          return input -> {
+            if (NullHandling.isNullOrEquivalent(input)) {
+              return false;
+            }
+            final int lowerComparing = stringComparator.compare(input, lowerBound);
+            return lowerComparing >= 0;
+          };
+        }
+      } else {
+        return Predicates.notNull();
+      }
+    });
+  }
+
+  private Predicate<Object[]> makeArrayPredicate(@Nullable TypeSignature<ValueType> arrayType)
+  {
```

Comment:
the same filter object is shared between all the segments being processed so it is probably worth not recomputing the predicate each time, can't use a supplier here since it is dependent on the input type, but i can make a map for the non-null input types, and a supplier for the per row detection

Reply:
changed to use a map for the array types and a separate supplier for per row detection